### PR TITLE
Fixes an issue that prevented using Groovy scripts from classpath in authentication

### PR DESCRIPTION
**Pull request by fcolin-odigo**
When defining a Groovy script from classpath from this configuration key `key cas.authn.policy.groovy[0].script=` (see [here](https://apereo.github.io/cas/6.5.x/authentication/Configuring-Authentication-Policy-Groovy.html) for details), the script cannot be loaded.

This was due to `org.apereo.cas.authentication.policy.GroovyScriptAuthenticationPolicy#initializeWatchableScriptIfNeeded()` that tried to create the resource from only a part of the provided URI (`classpath:/script.groovy` was loaded as `/script.groovy`), thus the code was trying to load the script from filesystem instead of classpath.

I used the full URI to load the resource, and added a test case to cover this case.

### CODE REVIEW

File: core/cas-server-core-authentication-api/src/test/resources/GroovyAuthenticationPolicy.groovy [hunk 1]
```diff
@@ -0,0 +1,4 @@
+def shouldResumeOnFailure(Object[] args) {
```
Comment (mmoayyed) [inline core/cas-server-core-authentication-api/src/test/resources/GroovyAuthenticationPolicy.groovy:1]: This is not a valid authentication policy file. Please update the script to match the docs.

Comment (fcolin-odigo) [reply core/cas-server-core-authentication-api/src/test/resources/GroovyAuthenticationPolicy.groovy:1]: Hi,
No problem, but I have copied this file from the [GroovyScriptAuthenticationPolicyTests.java](https://github.com/apereo/cas/blob/master/core/cas-server-core-authentication-api/src/test/java/org/apereo/cas/authentication/policy/GroovyScriptAuthenticationPolicyTests.java) examples. Should I also update them ?

**Comment by fcolin-odigo** [comment]
Should I also create a merge request for this bug on the 6.5.x branch?

File: core/cas-server-core-authentication-api/src/test/resources/GroovyAuthenticationPolicy.groovy (hunk 1 above)
Comment (mmoayyed) [reply core/cas-server-core-authentication-api/src/test/resources/GroovyAuthenticationPolicy.groovy:1]: No. You should only update the script file.

**Comment by mmoayyed** [comment]
Yes but only if you want the fix to go into the 6.5.x branch

File: core/cas-server-core-authentication-api/src/test/resources/GroovyAuthenticationPolicy.groovy (hunk 1 above)
Comment (fcolin-odigo) [reply core/cas-server-core-authentication-api/src/test/resources/GroovyAuthenticationPolicy.groovy]: Done, the test file is now a copy from [the documentation example](https://apereo.github.io/cas/6.5.x/authentication/Configuring-Authentication-Policy-Groovy.html).

**Comment by mmoayyed** [comment]
@apereocas-bot runci

**Comment by fcolin-odigo** [comment]
Hello @mmoayyed ,

I have checked automatic check failures and I have two questions before fix them:

1. [The checkstyle fails on "wrong import order"](https://github.com/apereo/cas/runs/6222050346?check_suite_focus=true#step:8:1273) on a file I didn’t touch the imports. Should I still fix them?
2. Should I [fix the test Groovy file issues](https://app.codacy.com/gh/apereo/cas/pullRequest?prid=9330044) as it is just a copy from documentation for testing purpose?

**Comment by mmoayyed** [comment]
Thank you for updating the PR. You do not need to fix those issues. They will be taken care of separately.

**Pull request merged**